tests/sentry/api/serializers/test_project.py

```diff
@@ -155,49 +155,109 @@ def test_simple(self):
+            organization_id=self.organization.id,
             name='production',
         )
-
+        environment_1.add_project(self.project)
+        environment_1.save()
+        environment_2 = Environment.objects.create(
+            organization_id=self.organization.id,
+            name='staging',
+        )
+        environment_2.add_project(self.project)
+        environment_2.save()
         deploy = Deploy.objects.create(
-            environment_id=environment.id,
-            organization_id=organization.id,
-            release=release,
-            date_finished=date
+            environment_id=environment_1.id,
+            organization_id=self.organization.id,
+            release=self.release,
+            date_finished=self.date
         )
-
         ReleaseProjectEnvironment.objects.create(
-            project_id=project.id,
-            release_id=release.id,
-            environment_id=environment.id,
+            project_id=self.project.id,
+            release_id=self.release.id,
+            environment_id=environment_1.id,
             last_deploy_id=deploy.id
         )
 
-        result = serialize(project, user, ProjectSummarySerializer())
+    def test_simple(self):
+        result = serialize(self.project, self.user, ProjectSummarySerializer())
 
-        assert result['id'] == six.text_type(project.id)
-        assert result['name'] == project.name
-        assert result['slug'] == project.slug
-        assert result['firstEvent'] == project.first_event
+        assert result['id'] == six.text_type(self.project.id)
+        assert result['name'] == self.project.name
+        assert result['slug'] == self.project.slug
+        assert result['firstEvent'] == self.project.first_event
+        assert 'releases' in result['features']
+        assert result['platform'] == self.project.platform
+
+        assert result['latestDeploys'] == {
+            'production': {'dateFinished': self.date, 'version': self.release.version}
+        }
+        assert result['latestRelease'] == serialize(self.release)
+        assert result['environments'] == ['production', 'staging']
+
+    def test_no_enviroments(self):
+        # remove environments and related models
+        Deploy.objects.all().delete()
+        Release.objects.all().delete()
+        Environment.objects.all().delete()
+
+        result = serialize(self.project, self.user, ProjectSummarySerializer())
+
+        assert result['id'] == six.text_type(self.project.id)
+        assert result['name'] == self.project.name
+        assert result['slug'] == self.project.slug
+        assert result['firstEvent'] == self.project.first_event
+        assert 'releases' in result['features']
+        assert result['platform'] == self.project.platform
+
+        assert result['latestDeploys'] is None
+        assert result['latestRelease'] is None
+        assert result['environments'] is None
+
+    def test_avoid_hidden_and_no_env(self):
```

Comment:
Added additional test to show we are not including no environment or hidden environments.

Reply:
🎉 Moar tests!